docs/src/man/reshaping_new.md

```diff
@@ -0,0 +1,498 @@
+  12 │  2021  Oil        12     26
+
+julia> costs = select(grocessary, :year, :list, [:month, :Date] .=> x -> x/2,
+                             renamecols=false)
+12×4 DataFrame
+ Row │ year   list    month    Date
+     │ Int64  String  Float64  Float64
+─────┼─────────────────────────────────
+   1 │  2019  Sugar       0.5      7.5
+   2 │  2019  Soap        1.0      8.0
+   3 │  2019  Fruits      1.5      8.5
+   4 │  2019  Oil         2.0      9.0
+   5 │  2020  Sugar       2.5      9.5
+   6 │  2020  Soap        3.0     10.0
+   7 │  2020  Fruits      3.5     10.5
+   8 │  2020  Oil         4.0     11.0
+   9 │  2021  Sugar       4.5     11.5
+  10 │  2021  Soap        5.0     12.0
+  11 │  2021  Fruits      5.5     12.5
+  12 │  2021  Oil         6.0     13.0
+```
+
+Now before moving further let's practice a bit basic data transformation skills:
+
+```jldoctest dataframe
+julia> long_grocessary = stack(grocessary, [:month, :Date], [:year, :list],
+                          variable_name=:days, value_name=:grocessary) 
+24×4 DataFrame
+ Row │ year   list    days    grocessary
+     │ Int64  String  String  Int64
+─────┼───────────────────────────────────
+   1 │  2019  Sugar   month            1
+   2 │  2019  Soap    month            2
+   3 │  2019  Fruits  month            3
+   4 │  2019  Oil     month            4
+   5 │  2020  Sugar   month            5
+   6 │  2020  Soap    month            6
+   7 │  2020  Fruits  month            7
+   8 │  2020  Oil     month            8
+  ⋮  │   ⋮      ⋮       ⋮         ⋮
+  18 │  2020  Soap    Date            20
+  19 │  2020  Fruits  Date            21
+  20 │  2020  Oil     Date            22
+  21 │  2021  Sugar   Date            23
+  22 │  2021  Soap    Date            24
+  23 │  2021  Fruits  Date            25
+  24 │  2021  Oil     Date            26
+                           9 rows omitted
+
+julia> long_costs = stack(costs, [:month, :Date], [:year, :list],
+                                 variable_name=:days, value_name=:costs)
+24×4 DataFrame
+ Row │ year   list    days    costs
+     │ Int64  String  String  Float64
+─────┼────────────────────────────────
+   1 │  2019  Sugar   month       0.5
+   2 │  2019  Soap    month       1.0
+   3 │  2019  Fruits  month       1.5
+   4 │  2019  Oil     month       2.0
+   5 │  2020  Sugar   month       2.5
+   6 │  2020  Soap    month       3.0
+   7 │  2020  Fruits  month       3.5
+   8 │  2020  Oil     month       4.0
+  ⋮  │   ⋮      ⋮       ⋮        ⋮
+  18 │  2020  Soap    Date       10.0
+  19 │  2020  Fruits  Date       10.5
+  20 │  2020  Oil     Date       11.0
+  21 │  2021  Sugar   Date       11.5
+  22 │  2021  Soap    Date       12.0
+  23 │  2021  Fruits  Date       12.5
+  24 │  2021  Oil     Date       13.0
+                        9 rows omitted
+
+julia> long = innerjoin(long_grocessary, long_costs, on=[:year, :list, :days])
+24×5 DataFrame
+ Row │ year   list    days    grocessary  costs
```

Comment:
it is not clear for me - from the interpretation point of view - why `:days` combines both `month` and `Date`